### PR TITLE
feat: set/edit secrets for data connectors

### PR DESCRIPTION
**Pull request by ciyer**
/deploy renku-data-services=leafty/feat-namespaced-data-connectors-5

### CODE REVIEW

File: client/src/features/projectsV2/api/data-connectors.enhanced-api.ts [hunk 1]
```diff
@@ -47,6 +58,26 @@ const injectedApi = api.injectEndpoints({
...
+        for (const dataConnectorId of dataConnectorIds) {
+          const response = await fetchWithBQ(
+            `/data_connectors/${dataConnectorId}/secrets`
+          );
+          if (response.error) {
+            return response;
+          }
+          result[dataConnectorId] =
+            response.data as GetDataConnectorsByDataConnectorIdSecretsApiResponse;
+        }
```
Comment (leafty) [inline client/src/features/projectsV2/api/data-connectors.enhanced-api.ts:77]: Maybe you can use `Promise.all()` or `Promise.allSettled()` to have the requests go in parallel? The current loop does them in sequence.

File: tests/cypress/e2e/groupV2DataSourceCredentials.spec.ts [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+
+    // Credentials should be changed
+    cy.getDataCy("data-connector-title").should(
+      "contain.text",
+      "example storage"
+    );
+    cy.getDataCy("access_key_id-value").should("contain.text", "<sensitive>");
+  });
+
+  //   it("edit a data source with credentials", () => {
```
Comment (leafty) [inline tests/cypress/e2e/groupV2DataSourceCredentials.spec.ts:304]: What is the intention with the commented out tests?

**Review by leafty — CHANGES_REQUESTED** [review]
Code mostly looks good and it seems to work. Some comments below:

File: tests/cypress/e2e/groupV2DataSourceCredentials.spec.ts (hunk 2 above)
Comment (ciyer) [reply tests/cypress/e2e/groupV2DataSourceCredentials.spec.ts]: I have cleaned up the test file in 6ce632ea6c73ac77e31375052c8bae442ca79c96.

**Review by leafty — APPROVED** [review]
Nice!

**Pull request merged**